### PR TITLE
docs: add release checklist

### LINKED ISSUE #120: [Feature] Harden release checklist and post-release verification
## Summary

Improve the PawWork release process so future releases are easier to run and verify.

## Scope

- Add a tracked release checklist.
- Require release notes before publishing a GitHub Release.
- Add post-release verification for user-facing downloads and updater metadata.
- Script or document the release workflow commands to reduce manual copy/paste mistakes.

## Checklist Location

Add the reusable checklist at `.github/RELEASE_CHECKLIST.md`.

`docs/` is local-only in this repo, so it is not a good place for tracked release process docs. `.github/` is better because this is GitHub Release and Actions workflow guidance.

## Release Notes Format

Document the v0.2.5 format as the default:

- English first.
- Direct Downloads links for the user-facing installers.
- A short Chinese section below the English notes.
- Do not rely on the GitHub Assets list as the primary download UI.

## Post-Release Verification

The release checklist or helper should verify:

- The GitHub Release is not a draft.
- The GitHub Release is not a prerelease for stable channels.
- The macOS Apple Silicon `.dmg` exists.
- The macOS Intel `.dmg` exists.
- The Windows x64 `.exe` exists.
- `latest.yml` points to the Windows installer.
- `latest-mac.yml` includes both macOS arm64 and x64 update entries.

## Non-Goals

- Do not repair v0.2.5 `latest-mac.yml`; ship the fixed updater metadata path in the next release.
- Do not remove `.zip`, `.blockmap`, or `latest*.yml` assets until updater requirements are proven safe.
- Do not redesign the release workflow beyond the checklist, verification, and command-helper scope.

## Acceptance Criteria

- `.github/RELEASE_CHECKLIST.md` exists and covers version bump, PR merge, workflow runs, release notes, publish step, post-release checks, and issue closeout.
- The checklist documents the English-first plus Chinese-summary release notes format.
- A post-release verification path exists, either as a script or as exact commands in the checklist.
- The release workflow command sequence for macOS submit/finalize and Windows full builds is scripted or documented enough to avoid manual parameter mistakes.

### PR DESCRIPTION
**Pull request by Astro-Han**
## Summary

Add a tracked release checklist and a post-release verification helper for PawWork desktop releases.

## Why

The v0.2.5 release exposed process gaps around release notes timing, manual release command handling, and updater metadata verification. This PR documents the intended release flow and adds a script that verifies user-facing installers plus updater metadata before release issues are closed.

## Related Issue

Closes #120

## How To Verify

```bash
cd packages/desktop-electron && bun test scripts/verify-release.test.ts
cd packages/desktop-electron && bun run typecheck
cd packages/desktop-electron && bun test
git diff --check
bun packages/desktop-electron/scripts/verify-release.ts v0.2.5
```

Expected result for the final command: it fails on v0.2.5 because that release's current `latest-mac.yml` does not include `pawwork-mac-arm64.zip`. That is the historical release gap this checklist/helper is meant to catch for future releases.

## Screenshots or Recordings

Not included. This is a release process and verification script change with no visible UI.

## Checklist

- [x] I linked the related issue, or stated why there is no issue
- [x] This PR has type, scope, and priority labels, or I requested maintainer labeling
- [x] I listed the relevant verification steps, including tests when behavior changed
- [x] I manually checked visible UI or copy changes when needed, with screenshots or recordings
- [x] I considered macOS and Windows impact for desktop, packaging, updater, signing, paths, shell, or permissions changes
- [x] I called out docs, release notes, dependencies, permissions, credentials, deletion behavior, or generated/local file changes when relevant
- [x] I am targeting `dev`, and my PR title and commit messages use Conventional Commits in English

## Summary by CodeRabbit

* **Documentation**
  * Added a stable release checklist documenting end-to-end desktop release steps (preflight, build, notarization, publish, and post-release verification), with English and Chinese release-note guidance.

* **Tests**
  * Added comprehensive tests covering release verification logic and many failure scenarios to ensure artifact and metadata integrity.

* **Chores**
  * Added a release verification utility to validate release state, required installers/sidecars, updater pointers, and tag formatting before publishing.

### CODE REVIEW

**Comment by Astro-Han** [comment]
@coderabbitai review

File: packages/desktop-electron/scripts/verify-release.ts [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+  for (const line of source.split(/\r?\n/)) {
+    const fileMatch = line.match(/^\s*-\s+url:\s*(.+?)\s*$/)
+    if (fileMatch) {
+      urls.push(fileMatch[1])
+      continue
+    }
+
+    const pathMatch = line.match(/^path:\s*(.+?)\s*$/)
+    if (pathMatch) urls.push(pathMatch[1])
```
[the remark this answers is not in the document]
Comment (Astro-Han) [reply packages/desktop-electron/scripts/verify-release.ts]: Fixed the quoted YAML scalar case by stripping matching single or double quotes after parsing the url/path value, and added a quoted metadata test. I did not add optional chaining on match groups because those accesses are already guarded by the match result; keeping them as direct indexed reads preserves the function string[] contract without introducing undefined values.

File: packages/desktop-electron/scripts/verify-release.ts [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+async function fetchText(url: string) {
+  const response = await fetch(url, { headers: githubHeaders() })
+  if (!response.ok) throw new Error(`Failed to fetch ${url}: ${response.status} ${response.statusText}`)
+  return response.text()
+}
+
+async function fetchJson<T>(url: string) {
+  const response = await fetch(url, { headers: githubHeaders() })
+  if (!response.ok) throw new Error(`Failed to fetch ${url}: ${response.status} ${response.statusText}`)
```
[the remark this answers is not in the document]
Comment (Astro-Han) [reply packages/desktop-electron/scripts/verify-release.ts]: Fixed. main() now wraps the GitHub API and asset download path in a try/catch and prints a concise failure message instead of surfacing an unhandled stack trace.

File: packages/desktop-electron/scripts/verify-release.ts [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    process.exit(2)
+  }
+
+  const repo = process.argv[3] ?? DEFAULT_REPO
+  const normalizedTag = normalizeTag(tag)
+  const release = await fetchJson<GithubRelease>(
+    `https://api.github.com/repos/${repo}/releases/tags/${encodeURIComponent(normalizedTag)}`,
+  )
+  const latestYml = await fetchText(findAsset(release, "latest.yml").browser_download_url)
+  const latestMacYml = await fetchText(findAsset(release, "latest-mac.yml").browser_download_url)
```
[the remark this answers is not in the document]
Comment (Astro-Han) [reply packages/desktop-electron/scripts/verify-release.ts]: Fixed. Missing latest.yml or latest-mac.yml no longer throws before verification. The script now skips the metadata download when the asset is missing and lets verifyReleasePayload report the missing asset plus the missing updater entries in one clean failure report. Added coverage for this case.

File: packages/desktop-electron/scripts/verify-release.ts [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+  for (const line of source.split(/\r?\n/)) {
+    const fileMatch = line.match(/^\s*-\s+url:\s*(.+?)\s*$/)
+    if (fileMatch) {
+      urls.push(fileMatch[1])
+      continue
+    }
+
+    const pathMatch = line.match(/^path:\s*(.+?)\s*$/)
+    if (pathMatch) urls.push(pathMatch[1])
+  }
```
[the remark this answers is not in the document]
Comment (Astro-Han) [reply packages/desktop-electron/scripts/verify-release.ts:51]: Fixed in 2171400. parseUpdaterFileUrls now normalizes captured YAML scalar values by stripping inline comments outside quotes, preserving # inside quoted filenames, and then removing matching outer quotes. The test now covers quoted url/path values with inline comments.

**Comment by Astro-Han** [comment]
@coderabbitai review

**Comment by Astro-Han** [comment]
@coderabbitai review

File: packages/desktop-electron/scripts/verify-release.ts [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+    const fileMatch = line.match(/^\s*-\s+url:\s*(.+?)\s*$/)
+    if (fileMatch) {
+      urls.push(parseYamlScalar(fileMatch[1]))
+      continue
+    }
+
+    const pathMatch = line.match(/^\s*path:\s*(.+?)\s*$/)
+    if (pathMatch) urls.push(parseYamlScalar(pathMatch[1]))
+  }
+
```
Comment (Astro-Han) [inline packages/desktop-electron/scripts/verify-release.ts:52]: The \`parseUpdaterFileUrls\` function uses regex to parse YAML, which is fragile. YAML allows more complex quoting and escaping patterns that these regexes wont handle correctly (e.g., multiline strings, escaped quotes within quoted strings, block scalars). Consider using a proper YAML parser like \`yaml\` or \`js-yaml\`, or at least document in a comment why this regex approach is chosen and what edge cases it deliberately ignores.

File: packages/desktop-electron/scripts/verify-release.ts [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+      quote = quote === char ? undefined : quote ?? char
+      continue
+    }
+
+    if (!quote && char === "#" && /\s/.test(value[index - 1] ?? " ")) {
+      return value.slice(0, index)
+    }
+  }
+
+  return value
```
Comment (Astro-Han) [inline packages/desktop-electron/scripts/verify-release.ts:79]: The \`stripInlineComment\` function assumes that \`#\` must be preceded by whitespace to start a comment, but YAML 1.2 spec allows comments to start at the beginning of a line without preceding whitespace. The check \`/\\s/.test(value[index - 1] ?? \` \`)\` handles the start-of-line case by defaulting to a space, which is correct, but this implicit behavior is subtle and worth documenting explicitly.

File: packages/desktop-electron/scripts/verify-release.ts [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+    const latestYml = await fetchAssetText(release, "latest.yml")
+    const latestMacYml = await fetchAssetText(release, "latest-mac.yml")
+    const failures = verifyReleasePayload({ release, latestYml, latestMacYml })
+
+    if (failures.length) {
+      console.error(`Release verification failed for ${repo} ${normalizedTag}:`)
+      for (const failure of failures) console.error(`- ${failure}`)
+      process.exit(1)
+    }
+
```
Comment (Astro-Han) [inline packages/desktop-electron/scripts/verify-release.ts:222]: When \`findAsset\` returns \`undefined\`, \`fetchAssetText\` returns an empty string. This conflates "asset does not exist" with "asset exists but is empty". In \`verifyReleasePayload\`, missing assets are already detected via \`REQUIRED_ASSETS\`, but for other checks this ambiguity could cause confusing error messages. Consider throwing or returning \`null\`/ \`undefined\` instead of an empty string to distinguish the two cases.

File: packages/desktop-electron/scripts/verify-release.test.ts [hunk 8]
```diff
@@ -0,0 +1,137 @@
+import { describe, expect, test } from "bun:test"
```
Comment (Astro-Han) [inline packages/desktop-electron/scripts/verify-release.test.ts]: Missing test coverage: (1) \`normalizeTag\` function is not tested - does it handle \`v0.2.6\` vs \`0.2.6\` correctly? What about malformed inputs like \`vv0.2.6\`? (2) \`prerelease\` flag detection - test suite only covers \`draft\`, not \`prerelease\`. (3) Missing DMG or blockmap assets - tests only cover missing \`.zip\` and \`latest*.yml\` assets. (4) Malformed YAML content - what happens if \`latest-mac.yml\` is corrupted or has syntax errors?

File: .github/RELEASE_CHECKLIST.md [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+- vX.Y.Z is published as the latest stable release.
+
+## 中文版本
+
+### 下载
+
+- [macOS Apple 芯片](https://github.com/Astro-Han/pawwork/releases/download/vX.Y.Z/pawwork-mac-arm64.dmg)
+- [macOS Intel 芯片](https://github.com/Astro-Han/pawwork/releases/download/vX.Y.Z/pawwork-mac-x64.dmg)
+- [Windows](https://github.com/Astro-Han/pawwork/releases/download/vX.Y.Z/pawwork-win-x64.exe)
+
```
Comment (Astro-Han) [inline .github/RELEASE_CHECKLIST.md:57]: The finalize workflow commands are long and easy to mis-copy. The instruction "Record each submit runs source run ID, source run attempt, source ref, source sha, workflow ref, workflow sha, and Apple submission ID from the workflow summary" is critical but could be clearer: where exactly in the workflow summary output are these values found? A sample workflow summary excerpt showing where to extract each value would make this less error-prone.

File: packages/desktop-electron/scripts/verify-release.ts [hunk 10]
```diff
@@ -0,0 +1,185 @@
...
+}
+
+async function fetchText(url: string) {
+  const response = await fetch(url, { headers: githubHeaders() })
+  if (!response.ok) throw new Error(`Failed to fetch ${url}: ${response.status} ${response.statusText}`)
+  return response.text()
+}
+
+async function fetchJson<T>(url: string) {
+  const response = await fetch(url, { headers: githubHeaders() })
```
Comment (Astro-Han) [inline packages/desktop-electron/scripts/verify-release.ts]: The \`normalizeTag\` function only adds a \`v\` prefix if missing, but does not validate the tag format. What happens if the user passes \`vv0.2.6\` or \`abc\`? The function would return \`vvv0.2.6\` or \`vabc\`, which would then fail at the GitHub API level with a 404. Consider adding a validation step or at least a warning for malformed inputs.

File: packages/desktop-electron/scripts/verify-release.ts [hunk 11]
```diff
@@ -0,0 +1,185 @@
...
+  if (!response.ok) throw new Error(`Failed to fetch ${url}: ${response.status} ${response.statusText}`)
+  return response.json() as Promise<T>
+}
+
+function findAsset(release: GithubRelease, name: string) {
+  return release.assets.find((entry) => entry.name === name)
+}
+
```
Comment (Astro-Han) [inline packages/desktop-electron/scripts/verify-release.ts:190]: The script reads \`GH_TOKEN\` from environment for authentication, but the PR description and checklist do not mention this requirement. Users running the verification script without \`GH_TOKEN\` will hit rate limits on public GitHub API (60 requests/hour for unauthenticated vs 5000 for authenticated). The checklist should mention setting \`GH_TOKEN\` or the script should print a warning when the token is missing.

File: .github/RELEASE_CHECKLIST.md [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+## Verification
+
+- macOS Apple Silicon submit/finalize completed successfully, including notarization.
+- macOS Intel submit/finalize completed successfully, including notarization.
+- Windows x64 release build completed successfully.
+- vX.Y.Z is published as the latest stable release.
+
+## 中文版本
+
```
Comment (Astro-Han) [inline .github/RELEASE_CHECKLIST.md:51]: The checklist states "Do not rely on the GitHub Assets list as the primary download UI" but does not explain why. Users may wonder: is it because the assets list does not show \`.dmg\` files prominently? Is it because updater metadata files clutter the list? Adding a brief rationale would make this instruction more convincing and easier to follow.

File: packages/desktop-electron/scripts/verify-release.ts [hunk 13]
```diff
@@ -0,0 +1,185 @@
...
+  browser_download_url: string
+}
+
+export type GithubRelease = {
+  tag_name: string
+  draft: boolean
+  prerelease: boolean
+  assets: GithubAsset[]
+}
+
```
Comment (Astro-Han) [inline packages/desktop-electron/scripts/verify-release.ts:13]: The \`GithubRelease\` type includes \`draft\` and \`prerelease\` fields, but the GitHub API response for releases also includes other fields like \`name\`, \`body\`, \`created_at\`, \`published_at\`. If future verification needs to check release notes content or timestamps, the type would need to be extended. Consider documenting that this is a minimal subset intentionally, or add a comment explaining the types scope.

File: packages/desktop-electron/scripts/verify-release.ts [hunk 14]
```diff
@@ -0,0 +1,185 @@
...
+  return release.assets.find((entry) => entry.name === name)
+}
+
+async function fetchAssetText(release: GithubRelease, name: string) {
+  const asset = findAsset(release, name)
+  if (!asset) return ""
+  return fetchText(asset.browser_download_url)
+}
+
+async function main() {
```
Comment (Astro-Han) [inline packages/desktop-electron/scripts/verify-release.ts:197]: The \`fetchText\` and \`fetchJson\` functions throw generic \`Error\` objects with only status code and message. This loses important debugging context: was this a network failure, authentication issue, rate limit, or GitHub API error? Consider creating custom error classes or at least including the URL and response headers in the error message for better debugging.

File: packages/desktop-electron/scripts/verify-release.ts (hunk 5 above)
Comment (Astro-Han) [reply packages/desktop-electron/scripts/verify-release.ts:52]: Documented the scope in the parser. I am not adding a YAML dependency here because the verifier only reads the fixed electron-builder url/path metadata fields, and this helper is deliberately not a general YAML parser. The comment now calls out the ignored cases like block scalars and multiline values.

File: packages/desktop-electron/scripts/verify-release.ts (hunk 7 above)
Comment (Astro-Han) [reply packages/desktop-electron/scripts/verify-release.ts:222]: Fixed. fetchAssetText now returns undefined for a missing release asset instead of an empty string, so a missing metadata asset stays distinguishable from an existing empty metadata file.

File: packages/desktop-electron/scripts/verify-release.ts (hunk 6 above)
Comment (Astro-Han) [reply packages/desktop-electron/scripts/verify-release.ts:79]: Fixed. Added an explicit comment that the fallback space is what makes a leading # behave as a YAML comment marker.

File: packages/desktop-electron/scripts/verify-release.test.ts (hunk 8 above)
Comment (Astro-Han) [reply packages/desktop-electron/scripts/verify-release.test.ts]: Fixed. Added coverage for normalizeTag with prefixed, unprefixed, and malformed tags; prerelease detection; missing DMG and blockmap assets; and malformed updater metadata that does not expose the required url/path entries.

File: packages/desktop-electron/scripts/verify-release.ts (hunk 10 above)
Comment (Astro-Han) [reply packages/desktop-electron/scripts/verify-release.ts]: Fixed. normalizeTag now validates the final tag format and fails locally for malformed inputs like vv0.2.6 or abc instead of sending them to the GitHub API as confusing 404s. Added tests for both valid forms and malformed inputs.

File: .github/RELEASE_CHECKLIST.md (hunk 12 above)
Comment (Astro-Han) [reply .github/RELEASE_CHECKLIST.md:51]: Fixed. Added the rationale: the GitHub Assets list mixes user installers with updater metadata, so direct download links are clearer for users.

File: packages/desktop-electron/scripts/verify-release.ts (hunk 11 above)
Comment (Astro-Han) [reply packages/desktop-electron/scripts/verify-release.ts:190]: Fixed. The checklist now recommends exporting GH_TOKEN from gh auth token before running the verifier, and the script prints a warning when GH_TOKEN is absent so operators know they are using the unauthenticated rate limit.

File: .github/RELEASE_CHECKLIST.md (hunk 9 above)
Comment (Astro-Han) [reply .github/RELEASE_CHECKLIST.md:57]: Fixed. Added a sample submit workflow summary excerpt showing the source_run_id, source_run_attempt, source_ref, source_sha, source_workflow_ref, source_workflow_sha, and submission_id fields to copy into finalize.

File: packages/desktop-electron/scripts/verify-release.ts (hunk 14 above)
Comment (Astro-Han) [reply packages/desktop-electron/scripts/verify-release.ts:197]: Fixed without adding custom error classes. fetchText and fetchJson now include the URL plus GitHub rate limit headers when available, which covers the practical debugging cases for this CLI while keeping the error path simple.

File: packages/desktop-electron/scripts/verify-release.ts (hunk 13 above)
Comment (Astro-Han) [reply packages/desktop-electron/scripts/verify-release.ts:13]: Fixed. Added a short comment that GithubRelease is intentionally a minimal GitHub Release API subset used by this verifier, so future release-note or timestamp checks can extend it when needed.

**Comment by Astro-Han** [comment]
@coderabbitai review

File: packages/desktop-electron/scripts/verify-release.ts [hunk 15]
```diff
@@ -0,0 +1,204 @@
...
+
+function parseYamlScalar(value: string) {
+  const trimmed = stripInlineComment(value).trim()
+  const quote = trimmed[0]
+  if ((quote === `"` || quote === `'`) && trimmed.at(-1) === quote) return trimmed.slice(1, -1)
+  return trimmed
+}
+
+function stripInlineComment(value: string) {
+  let quote: string | undefined
```
Comment (Astro-Han) [inline packages/desktop-electron/scripts/verify-release.ts:64]: This manual YAML parser is not a general-purpose solution and intentionally ignores block scalars (`|`, `>`) and multiline values. Consider adding a comment noting the limitations for future maintainers, or alternatively, use a proper YAML parser library (e.g., `yaml` package) for robustness.

File: packages/desktop-electron/scripts/verify-release.ts [hunk 16]
```diff
@@ -0,0 +1,204 @@
...
+      return value.slice(0, index)
+    }
+  }
+
+  return value
+}
+
+function hasUpdaterEntry(urls: string[], expected: string) {
+  return urls.some((url) => url === expected || url.endsWith(`/${expected}`))
+}
```
Comment (Astro-Han) [inline packages/desktop-electron/scripts/verify-release.ts:93]: The escape handling `value[index - 1] !== "\\"` checks for backslash escaping, but does not handle double-escaped backslashes before quotes (e.g., `\\"` should end the quote, but `\\\\"` should not). Consider whether electron-builder metadata ever produces such edge cases, and if so, add a test case for it.

File: packages/desktop-electron/scripts/verify-release.ts [hunk 17]
```diff
@@ -0,0 +1,204 @@
...
+  if (process.env.GH_TOKEN) headers.set("Authorization", `Bearer ${process.env.GH_TOKEN}`)
+  return headers
+}
+
+async function fetchText(url: string) {
+  const response = await fetch(url, { headers: githubHeaders() })
+  if (!response.ok) throw new Error(formatFetchError("fetch", url, response))
+  return response.text()
+}
+
```
Comment (Astro-Han) [inline packages/desktop-electron/scripts/verify-release.ts:154]: Network requests lack timeout handling. A hung request could block the verification script indefinitely. Consider adding \`AbortSignal.timeout()\` or a wrapped fetch with configurable timeout, especially since GitHub API can occasionally be slow.

File: packages/desktop-electron/scripts/verify-release.ts [hunk 18]
```diff
@@ -0,0 +1,204 @@
...
+async function main() {
+  try {
+    const tag = process.argv[2]
+    if (!tag) {
+      console.error("Usage: bun packages/desktop-electron/scripts/verify-release.ts <tag> [owner/repo]")
+      process.exit(2)
+    }
+
+    const repo = process.argv[3] ?? DEFAULT_REPO
```
Comment (Astro-Han) [inline packages/desktop-electron/scripts/verify-release.ts:205]: When \`findAsset\` returns \`undefined\`, this function returns \`undefined\`. But if the asset exists and \`fetchText\` throws (network error, server error), the exception propagates directly. This inconsistency means callers cannot distinguish "asset missing" from "asset exists but fetch failed". Consider returning a result type that captures both states, or throwing a consistent error type.

File: .github/RELEASE_CHECKLIST.md [hunk 19]
```diff
@@ -0,0 +1,141 @@
...
+The submit workflow summary should include values like this:
+
+```text
+source_run_id: 123456789
+source_run_attempt: 1
+source_ref: dev
+source_sha: 0123456789abcdef0123456789abcdef01234567
+source_workflow_ref: dev
+source_workflow_sha: 0123456789abcdef0123456789abcdef01234567
```
Comment (Astro-Han) [inline .github/RELEASE_CHECKLIST.md:84]: This placeholder naming (\`ARM64_SOURCE_RUN_ID\` etc.) is descriptive but the instruction "Record each submit runs source run ID..." earlier in the checklist does not clearly link these recorded values to these placeholder names. Consider adding a direct mapping example like: "Replace ARM64_SOURCE_RUN_ID with the value recorded from the arm64 submit workflow summary."

File: .github/RELEASE_CHECKLIST.md [hunk 20]
```diff
@@ -0,0 +1,141 @@
...
+
+```md
+## Downloads
+
+- [macOS Apple Silicon](https://github.com/Astro-Han/pawwork/releases/download/vX.Y.Z/pawwork-mac-arm64.dmg)
+- [macOS Intel](https://github.com/Astro-Han/pawwork/releases/download/vX.Y.Z/pawwork-mac-x64.dmg)
+- [Windows](https://github.com/Astro-Han/pawwork/releases/download/vX.Y.Z/pawwork-win-x64.exe)
+
+## Highlights
+
```
Comment (Astro-Han) [inline .github/RELEASE_CHECKLIST.md:36]: The \`...\` placeholders are ambiguous. A maintainer may wonder what content belongs in "Highlights" vs "Runtime And Maintenance". Consider adding inline examples or a comment clarifying typical content for each section.

File: .github/RELEASE_CHECKLIST.md [hunk 21]
```diff
@@ -0,0 +1,141 @@
...
+gh workflow run build.yml --repo Astro-Han/pawwork --ref ARM64_SOURCE_WORKFLOW_REF -f phase=finalize -f channel=prod -f arch=arm64 -f source_run_id=ARM64_SOURCE_RUN_ID -f source_run_attempt=ARM64_SOURCE_RUN_ATTEMPT -f source_ref=ARM64_SOURCE_REF -f source_sha=ARM64_SOURCE_SHA -f source_workflow_ref=ARM64_SOURCE_WORKFLOW_REF -f source_workflow_sha=ARM64_SOURCE_WORKFLOW_SHA -f submission_id=ARM64_SUBMISSION_ID
+gh workflow run build.yml --repo Astro-Han/pawwork --ref X64_SOURCE_WORKFLOW_REF -f phase=finalize -f channel=prod -f arch=x64 -f source_run_id=X64_SOURCE_RUN_ID -f source_run_attempt=X64_SOURCE_RUN_ATTEMPT -f source_ref=X64_SOURCE_REF -f source_sha=X64_SOURCE_SHA -f source_workflow_ref=X64_SOURCE_WORKFLOW_REF -f source_workflow_sha=X64_SOURCE_WORKFLOW_SHA -f submission_id=X64_SUBMISSION_ID
+```
+
+Build and publish the Windows installer:
+
+```bash
+gh workflow run build.yml --repo Astro-Han/pawwork --ref dev -f phase=full -f channel=prod -f target=windows -f arch=x64
+```
+
```
Comment (Astro-Han) [inline .github/RELEASE_CHECKLIST.md:102]: The checklist describes running verification but does not explain remediation steps when verification fails. Consider adding a brief section like: "If verification fails: 1) Check which assets are missing or malformed; 2) Re-run the affected build phase; 3) Re-publish the release only after verification passes."

File: packages/desktop-electron/scripts/verify-release.test.ts [hunk 22]
```diff
@@ -0,0 +1,182 @@
...
```
Comment (Astro-Han) [inline packages/desktop-electron/scripts/verify-release.test.ts]: Test coverage is solid for parsing scenarios, but missing tests for network failure paths: API returning non-JSON, HTTP 4xx/5xx errors, and rate limit exhaustion. The \`fetchJson\` and \`fetchText\` error paths are exercised only implicitly through the main script, not via unit tests.

File: packages/desktop-electron/scripts/verify-release.test.ts [hunk 23]
```diff
@@ -0,0 +1,182 @@
...
+
+import { normalizeTag, parseUpdaterFileUrls, verifyReleasePayload, type GithubRelease } from "./verify-release"
+
+const baseRelease: GithubRelease = {
```
Comment (Astro-Han) [inline packages/desktop-electron/scripts/verify-release.test.ts:18]: The \`baseRelease\` fixture includes all required assets but does not test the edge case where \`browser_download_url\` is a full URL with path segments. \`hasUpdaterEntry\` handles this via \`endsWith\`, but no test verifies that behavior with URLs like \`https://github.com/.../releases/download/vX.Y.Z/pawwork-mac-arm64.zip\`.

File: packages/desktop-electron/scripts/verify-release.ts [hunk 24]
```diff
@@ -0,0 +1,204 @@
...
+  return url.split("/").at(-1) ?? url
+}
+
+function verifyReferencedAssets(sourceName: string, urls: string[], assetNames: Set<string>, failures: string[]) {
+  for (const url of urls) {
+    const asset = assetNameFromUrl(url)
+    if (!assetNames.has(asset)) failures.push(`${sourceName} references missing release asset: ${asset}`)
+  }
+}
+
```
Comment (Astro-Han) [inline packages/desktop-electron/scripts/verify-release.ts:105]: The \`url.endsWith(\`/${expected}\`)\` check assumes URLs contain a \`/\` separator. If a URL is just the filename (no path prefix), this check would fail even when the asset name matches exactly. The \`url === expected\` fallback covers this, but consider clarifying with a comment that both cases are intentional.

File: .github/RELEASE_CHECKLIST.md [hunk 25]
```diff
@@ -0,0 +1,141 @@
...
+Verify the draft release has all expected user-facing installers before publishing:
+
+```bash
+gh release view vX.Y.Z --repo Astro-Han/pawwork --json isDraft,isPrerelease,assets,url
+```
+
+Publish the release as the latest stable release:
+
+```bash
+gh release edit vX.Y.Z --repo Astro-Han/pawwork --draft=false --latest
```
[the remark this answers is not in the document]
Comment (Astro-Han) [reply .github/RELEASE_CHECKLIST.md]: Fixed. The publish command now explicitly uses `--prerelease=false` together with `--draft=false --latest`, and I verified the local `gh release edit` help supports boolean false syntax for release flags.

File: packages/desktop-electron/scripts/verify-release.ts [hunk 26]
```diff
@@ -0,0 +1,204 @@
...
+async function fetchText(url: string) {
+  const response = await fetch(url, { headers: githubHeaders() })
+  if (!response.ok) throw new Error(formatFetchError("fetch", url, response))
+  return response.text()
+}
+
+async function fetchJson<T>(url: string) {
+  const response = await fetch(url, { headers: githubHeaders() })
+  if (!response.ok) throw new Error(formatFetchError("fetch", url, response))
+  return response.json() as Promise<T>
```
[the remark this answers is not in the document]
Comment (Astro-Han) [reply packages/desktop-electron/scripts/verify-release.ts]: Fixed. GitHub API and asset fetches now go through a shared 15 second timeout using `AbortSignal.timeout`, and network failures include the request URL in the error message.

File: packages/desktop-electron/scripts/verify-release.ts (hunk 15 above)
Comment (Astro-Han) [reply packages/desktop-electron/scripts/verify-release.ts:64]: Fixed. The parser comment now says this helper is scoped to electron-builder `url` and `path` metadata, not general YAML, and explicitly lists ignored cases like block scalars and multiline values.

File: packages/desktop-electron/scripts/verify-release.ts (hunk 16 above)
Comment (Astro-Han) [reply packages/desktop-electron/scripts/verify-release.ts:93]: Fixed. Quote escaping now counts the run of preceding backslashes, so odd and even backslash counts are handled correctly. Added coverage for escaped quotes and double backslashes before a quote.

File: packages/desktop-electron/scripts/verify-release.ts (hunk 17 above)
Comment (Astro-Han) [reply packages/desktop-electron/scripts/verify-release.ts:154]: Fixed. Fetches now use the shared 15 second timeout wrapper, so a stalled GitHub API or asset request fails instead of hanging the verifier.

File: packages/desktop-electron/scripts/verify-release.ts (hunk 18 above)
Comment (Astro-Han) [reply packages/desktop-electron/scripts/verify-release.ts:205]: Not changed. Missing metadata asset and failed metadata download are intentionally different states here: a missing asset is release payload validation input and should be aggregated by `verifyReleasePayload`; an existing asset that cannot be fetched means the verifier could not run and should stop with the network/API error. A result type would add complexity without changing the operator action.

File: .github/RELEASE_CHECKLIST.md (hunk 19 above)
Comment (Astro-Han) [reply .github/RELEASE_CHECKLIST.md:84]: Fixed. Added direct mapping text from `ARM64_` and `X64_` placeholders to the submit workflow summary fields, including `ARM64_SOURCE_RUN_ID` from the arm64 `source_run_id` value.

File: .github/RELEASE_CHECKLIST.md (hunk 20 above)
Comment (Astro-Han) [reply .github/RELEASE_CHECKLIST.md:36]: Fixed. Replaced the ambiguous ellipses with concrete examples for Highlights, Runtime And Maintenance, and the Chinese summary section.

File: .github/RELEASE_CHECKLIST.md (hunk 21 above)
Comment (Astro-Han) [reply .github/RELEASE_CHECKLIST.md:102]: Fixed. Added remediation guidance for verification failures: inspect the reported missing or malformed asset, rerun the affected build phase, and publish only after the verifier passes.

File: packages/desktop-electron/scripts/verify-release.test.ts (hunk 22 above)
Comment (Astro-Han) [reply packages/desktop-electron/scripts/verify-release.test.ts]: Fixed. Added unit coverage for HTTP errors with rate-limit headers, network failures that include the request URL, and invalid JSON responses that include the request URL.

File: packages/desktop-electron/scripts/verify-release.test.ts (hunk 23 above)
Comment (Astro-Han) [reply packages/desktop-electron/scripts/verify-release.test.ts:18]: Fixed. Added a metadata test where updater entries are full GitHub download URLs with path segments, covering the `endsWith` branch.

File: packages/desktop-electron/scripts/verify-release.ts (hunk 24 above)
Comment (Astro-Han) [reply packages/desktop-electron/scripts/verify-release.ts:105]: Fixed. Added a comment documenting that updater metadata may contain either bare filenames or full download URLs, so both `url === expected` and `url.endsWith(...)` are intentional.
